src/controls/ObjectArrayControl.test.tsx

```diff
@@ -0,0 +1,143 @@
+import { test } from "vitest"
+import { screen, waitFor } from "@testing-library/react"
+import userEvent from "@testing-library/user-event"
+import { render } from "../common/test-render"
+
+import {
+  objectArrayControlUISchema,
+  objectArrayControlJsonSchema,
+  objectArrayControlUISchemaWithIcons,
+} from "../testSchemas/objectArraySchema"
+import { expect } from "@storybook/test"
+
+test("ObjectArrayControl renders without any data", async () => {
+  render({
+    schema: objectArrayControlJsonSchema,
+    uischema: objectArrayControlUISchema,
+  })
+  await screen.findByRole("button")
+  await screen.findByText("Add Assets")
+  await screen.findByText("No data")
+})
+
+test("ObjectArrayControl renders disabled remove button with one element", async () => {
+  render({
+    schema: objectArrayControlJsonSchema,
+    uischema: objectArrayControlUISchema,
+    data: { assets: [{ asset: "my asset" }] },
+  })
+  await screen.findByText("Add Assets")
+  await screen.findByDisplayValue("my asset")
+  //note: the text is within a span in the <button>
+  const removeButton = (await screen.findByText("Delete")).parentNode
+  expect(removeButton).toHaveProperty("disabled", true)
+})
+
+test("ObjectArrayControl renders enabled remove buttons with multiple elements", async () => {
+  render({
+    schema: objectArrayControlJsonSchema,
+    uischema: objectArrayControlUISchema,
+    data: { assets: [{ asset: "my asset" }, { asset: "my other asset" }] },
+  })
+  await screen.findByText("Add Assets")
+  await screen.findByDisplayValue("my asset")
+  await screen.findByDisplayValue("my other asset")
+  //note: the text is within a span in the <button>
+  const removeButton = (await screen.findAllByText("Delete"))[0].parentNode
```

Comment:
Did you try using findAllByRole instead? sometimes it's slow, but if it isn't causing problems, it's nice to use
```
const removeButton = await screen.findAllByRole("button", { name: "Delete" })[0]
```